docs/sources/policy-plugin.md

```diff
@@ -0,0 +1,214 @@
+# Policy Plugin
+
+This page explains how to implement your own policy plugin and how to configure
+it.
+
+The policy plugin enables to apply policy rules flexibly which are difficult
+to define with the [OpenConfig](http://www.openconfig.net/) policy model, just
+logging received paths and complex attributes manipulation for example.
+
+## Prerequisites
+
+The following assumes you finished [Getting Started](getting-started.md) and
+bases of [GoBGP Policy Model](policy.md).
+
+Also, the policy plugin feature uses ["plugin"](https://golang.org/pkg/plugin/)
+package of Golang, which requires to use Golang version 1.8 or later.
+
+## Contents
+
+- [Implementation of Policy Plugin](#implementation-of-policy-plugin)
+- [Build Policy Plugin](#build-policy-plugin)
+- [Configuration](#configuration)
+- [Verification](#verification)
+
+## Implementation of Policy Plugin
+
+A policy plugin must have an `Apply` function which receives `path *table.Path`
+and `options *table.PolicyOptions` arguments, then returns `*table.Path` and
+`error`.
+
+```go
+package main
+
+import (
+    "github.com/osrg/gobgp/table"
+)
+
+func Apply(path *table.Path, options *table.PolicyOptions) (*table.Path, error) {
+    // Do some manipulation
+    return path, nil
+}
+```
+
+`path *table.Path` includes the incoming path information and
+`options *table.PolicyOptions` provides the additional information about sender
+router for example.
+
+The return value of `*table.Path` should be the updated path if successfully
+manipulated (also with nothing to update) and `error` describes errors during
+manipulation process.
+
+Example: `policy_a.go`
+
+```go
+package main
+
+import (
+    "fmt"
+
+    "github.com/osrg/gobgp/table"
+)
+
+func Apply(path *table.Path, options *table.PolicyOptions) (*table.Path, error) {
+    // Just do logging
+    fmt.Printf("Apply policy to %+v\n", path)
+    return path, nil
+}
+```
+
+## Build Policy Plugin
+
+To generate a loadable policy plugin, execute `go build` command with
+`-buildmode=plugin` option.
+
+```bash
+go build -buildmode=plugin -o <plugin name>.so <plugin name>.go
+```
+
+Example: Build `policy_a.so` with `policy_a.go`
+
+```bash
+go build -buildmode=plugin -o policy_a.so policy_a.go
+```
+
+## Configuration
+
+To attach the policy plugin to global RIB or neighbor's local RIB (only when
+the neighbor is configured as a "route-server client"), specify path to the
+generated policy plugin (`.so` file) with `plugin-path` in
+`[[policy-definitions]]` section.
+
+Example: Attach `policy_a.so` to global RIB as "import policy"
```

Comment:
can you reload it at runtime if there is any change in the policy? How?

Reply:
Sorry for the delay. It’s a troublesome...
When reloading the confirmation, GoBGP will determine the configured values are updated or not before reload API. Then if the configuration file is not changed, even if the plug in is updated, GoBGP can’t detect change of the plug in's implementation.
So if we need to update and reload the plug in, it is required to change the path to plug in or its name.
For example, `policy_a.1.0.so` to `policy_a.1.1.so` or appending the updated timestamp or date.

Reply:
Thanks for the quick answers. I think renaming things and changing the path is fine. Might be worth adding a note to the README so people is aware this is how policies are updated when you are using this method.

Reply:
Yeah, thanks a lot! I will add description about how to update the policy plug-in and how to reject the given prefix.